File: Experiment/week14/page_calc.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
  if(argc != 2) // error case check first
  {
    printf("usage: page_calc [VIRTUAL ADDRESS NUMBER]\n\n");
    return 1;
  }

  // every var. cannot be negative
  unsigned int address = atoi(argv[1]); // address string to unsigned int
  unsigned int page_num = address / 4096; // location of page number
  unsigned int offset = address % 4096; // remainedr will be an offset

  printf("The address %u contains:\n", address);
  printf("Page Number: %u\n", page_num);
  printf("Offset: %u\n\n", offset);

  return 0;
}

File: Experiment/week14/page_calc.py
import sys

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("\nusage: page_calc [VIRTUAL ADDRESS NUMBER]\n\n")
        sys.exit(1)

    address = int(sys.argv[1])  # address string to unsigned int
    page_num = address >> 12    # location of page number
    offset = address & 0xfff    # remainedr will be an offset

    print("The address", address, "contains:")
    print("Page Number:", page_num)
    print("Offset:", offset)
